libs/internal/src/network/http_requester.cpp

```diff
@@ -108,20 +120,42 @@ HttpRequest::HttpRequest(std::string const& url,
     if (uri_components->has_port()) {
         port_ = uri_components->port();
     }
+
+    if (properties_.HttpProxy()) {
```

Comment:
Side note: it would be much better if we could validate these URLs earlier in the config process, but that would be a more major refactoring.